uk.co.compucorp.civicrm.hrleaveandabsences/api/v3/LeaveRequest.php

```diff
@@ -429,3 +429,198 @@ function civicrm_api3_leave_request_ismanagedby($params) {
+ *
+ * Uses the API exposed by the Comment entity (Comment.create)
+ * to create comments related to a LeaveRequest.
+ *
+ * @param $params
+ *
+ * @return array
+ */
+function civicrm_api3_leave_request_addcomment($params) {
+  _civicrm_api3_leave_request_prepare_parameters_for_comment_payload($params);
+  $results =  civicrm_api3('Comment', 'create', $params);
+
+  if($results['count'] > 0) {
+    _civicrm_api3_leave_request_filter_comment_return_data($results['values']);
+  }
+
+  return $results;
+}
+
+/**
+ * LeaveRequest.getComment API spec
+ *
+ * @param $spec
+ */
+function _civicrm_api3_leave_request_getcomment_spec(&$spec) {
+  $spec['comment_id'] = [
+    'name' => 'comment_id',
+    'type' => CRM_Utils_Type::T_INT,
+    'title' => 'Comment ID',
+    'description' => 'The Comment ID',
+    'api.required' => 0
+  ];
+
+  $spec['leave_request_id'] = [
+    'name' => 'leave_request_id',
+    'type' => CRM_Utils_Type::T_INT,
+    'title' => 'Leave Request ID',
+    'description' => 'The ID of the Leave Request',
+    'api.required' => 0,
+  ];
+
+  $spec['contact_id'] = [
+    'name' => 'contact_id',
+    'type' => CRM_Utils_Type::T_INT,
+    'title' => 'Contact (Commenter)',
+    'description' => 'The contact who made the comment',
+    'api.required' => 0,
+    'FKClassName'  => 'CRM_Contact_DAO_Contact',
+    'FKApiName'    => 'Contact',
+  ];
+}
+
+/**
+ * LeaveRequest.getComment API
+ *
+ * Uses the API exposed by the Comment entity (Comment.get)
+ * to fetch comments associated with a LeaveRequest
+ *
+ * @param $params
+ *
+ * @return array
+ */
+function civicrm_api3_leave_request_getcomment($params) {
+  _civicrm_api3_leave_request_prepare_parameters_for_comment_payload($params);
+  $results =  civicrm_api3('Comment', 'get', $params);
+
+  if($results['count'] > 0) {
+    _civicrm_api3_leave_request_filter_comment_return_data($results['values']);
+  }
+
+  return $results;
+}
+
+/**
+ * LeaveRequest.deleteComment API spec
+ *
+ * @param $spec
+ */
+function _civicrm_api3_leave_request_deletecomment_spec(&$spec) {
+  $spec['comment_id'] = [
+    'name' => 'id',
+    'type' => CRM_Utils_Type::T_INT,
+    'title' => 'Comment ID',
+    'description' => 'The Comment ID',
+    'api.required' => 1
+  ];
+}
+
+/**
+ * LeaveRequest.deleteComment API
+ *
+ * Uses the API exposed by the Comment entity (Comment.delete)
+ * to delete comments associated with a LeaveRequest.
+ * This method also implement some checks to ensure that only the LeaveRequest Approver
+ * or an Admin can delete a comment
+ *
+ * @param $params
+ *
+ * @return array
+ */
+function civicrm_api3_leave_request_deletecomment($params) {
+  _civicrm_api3_leave_request_prepare_parameters_for_comment_payload($params);
+
+  $comments =  civicrm_api3('Comment', 'get', ['id' => $params['id'], 'sequential' => 1]);
+
+  if ($comments['count'] > 0){
+    $leaveRequest = CRM_HRLeaveAndAbsences_BAO_LeaveRequest::findById($comments['values'][0]['entity_id']);
```

Comment:
If we don't pass the `entity_name` to the Comment.get API, we need to check in here that it's equal to 'LeaveRequest'. Otherwise we can end up deleting comments not beloging to LeaveRequest